packages/typespec-azure-resource-manager/src/rules/arm-delete-response-codes.ts

```diff
@@ -0,0 +1,45 @@
+import { Model, createRule } from "@typespec/compiler";
+
+import { getLroMetadata } from "@azure-tools/typespec-azure-core";
+import { getHttpOperation } from "@typespec/http";
+import { getArmResources } from "../resource.js";
+
+/**
+ * Verify that a delete operation only
+ */
+export const armDeleteResponseCodesRule = createRule({
+  name: "arm-delete-operation-response-codes",
+  severity: "warning",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/rules/delete-operation-response-codes",
+  description: "Ensure delete operations have the appropriate status codes.",
+  messages: {
+    sync: `Synchronous delete operations must have 200, 204 and default responses. They must not have any other responses.`,
+    async: `Long-running delete operations must have 202, 204 and default responses. They must not have any other responses.`,
+  },
```

Comment:
@markcowl would you prefer that I refactor this as a "response-codes" rule and have separate messages for delete/post/etc or make each it's own rule? I'm fine either way.

Reply:
Going to leave this as-is for now since code complete it approaching.

Reply:
I think we can add this in a separate PR